FinalEngine.ECS/Entity.cs

```diff
@@ -32,6 +32,17 @@ public Entity(Guid? uniqueID = null)
         this.typeToComponentMap = new Dictionary<Type, IComponent>();
     }
 
+    /// <summary>
+    /// Gets the components attached to this <see cref="Entity"/>.
+    /// </summary>
+    /// <value>
+    /// The components attached to this <see cref="Entity"/>.
+    /// </value>
+    public ICollection<IComponent> Components
```

Comment:
This should be internal. Just use the `InternalsvisibileTo` attribute to allow the view model project to have access to it. Users shouldn't have direct access to components as this is the responsibility of systems.